src/prommis/uky/uky_flowsheet.py

```diff
@@ -1467,31 +1467,193 @@ def add_costing(flowsheet):
     ]
 
     # define product flowrates
-    # TODO: Need to revisit this after getting costs for products defined in mw_comp_product in ree_feed_roaster
+    # TODO: Default sale prices for some components in flow_mol_comp_product are missing
+
+    ce_product = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Ce"]
+            * 328.24
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
```

Comment:
I have a few questions/comments here.

1. Why are you getting the value of the flowrate here? This means that this is a static value and is no longer coupled to the flowsheet.
2. Looking at the next section, you then take these static values and multiply by the units. You can get this directly by using the `as_quantity(var)` function instead of `value()` - this returns a `pint` `Quantity` which has both the magnitude and units.
3. Finally, I would be inclined to name these `Ce_product` as the capitalization is part of the name. This is mostly useful for things like `Y_product` which could be a bit more ambiguous.

Reply:
It should be able to do this, however I would not do it that way myself. `as_quantity` returns a `Quantity` so it us the units in it anyway so you can do the conversion after the fact.

However, this ties into the question overall of whether you should be using `value` or `as_quantity` at all.

Reply:
Also, can you show the rest of the traceback? That last line is coming from `pint` alone, so it could be a `pint` issue.

Reply:
I think we could just get away with not using `value `or `as_quantity`, but it's currently running into a DOF issue when I do this, so I need to figure out why.